auth_user_insensitive/models/res_users.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+
+    _inherit = 'res.users'
+
+    login = fields.Char(
+        'Login',
+        size=64,
+        required=True,
+        help='Used to log into the system. Case insensitive.',
+    )
+
+    @api.model
+    def search(self, domain, *args, **kwargs):
+        """ Overload search to lowercase name domains. This can't be done in
+        a regular search method as the field is not computed
```

Comment:
IMO it's more explicit to extend `_login` to use [`=ilike`](https://www.odoo.com/documentation/10.0/reference/orm.html#domains) instead of `=`. No ``search()` magic needed that way.
And this extension would run after and if the default `=` search fails.